kernel/include/kswap.h

```diff
@@ -113,11 +112,11 @@ static ALWAYS_INLINE unsigned int do_swap(unsigned int key,
 			z_smp_release_global_lock(new_thread);
 		}
 #endif
+		sys_trace_thread_switched_out();
 		_current_cpu->current = new_thread;
 		wait_for_switch(new_thread);
 		arch_switch(new_thread->switch_handle,
 			     &old_thread->switch_handle);
-		sys_trace_thread_switched_in();
```

Comment:
I'm curious, was this part of the problem for counting switched in twice? It looks like it, but wanted to ask.

Reply:
this is added again in a later patch, they will be sqaushed, but yes, we were tracing both in C and asm.